main/unwindi.c

```diff
@@ -257,37 +262,33 @@ extern MIOPos uwiGetFilePosition (void)
 
 extern void uwiPushMarker (void)
 {
-	Assert (uwiMarkerStack);
+	Assert (uwiCurrentMarker - uwiMarkerStack < 255);
 
-	if (uwiCurrentMarker)
-		ptrArrayAdd (uwiMarkerStack, uwiCurrentMarker);
+	if (uwiCurrentMarker) uwiCurrentMarker++;
+	else uwiCurrentMarker = uwiMarkerStack;
 
-	uwiCurrentMarker = ptrArrayNew ((ptrArrayDeleteFunc)uugcDeleteC);
+	*uwiCurrentMarker = 0;
 }
 
 extern void uwiPopMarker (const int upto, const bool revertChars)
 {
+	Assert (uwiCurrentMarker - uwiMarkerStack >= 0);
 	uwiClearMarker (upto, revertChars);
 
-	ptrArrayDelete (uwiCurrentMarker);
-
-	uwiCurrentMarker = NULL;
-	if (ptrArrayCount (uwiMarkerStack) > 0)
-	{
-		uwiCurrentMarker = ptrArrayLast (uwiMarkerStack);
-		ptrArrayRemoveLast (uwiMarkerStack);
-	}
+	if (uwiCurrentMarker == uwiMarkerStack) uwiCurrentMarker = NULL;
+	else uwiCurrentMarker--;
```

Comment:
If you introduce the code I showed above comment, push an pop calls become unbalanced.
As the result, `uwiCurrentMarker - uwiMarkerStack` can be less than 0.

Depending Assert in logic is a bad idea.

Let's handle the condition `>=0' explicitly here.

Reply:
Ok.